File: common/object/dto/PromiseResponse.go
package dto

import (
	"promise/common/object/constValue"
	"promise/common/object/model"
	"time"
)

// PromiseResponseInterface is the APIs that a DTO should have.
type PromiseResponseInterface interface {
	Load(data interface{}) error
	GetID() string
	GetURI() string
	GetCategory() string
	GetCreatedAt() time.Time
	GetUpdatedAt() time.Time
}

// PromiseResponse is the base type of a response in Promise.
type PromiseResponse struct {
	ID        string    `json:"ID"`
	URI       string    `json:"URI"`
	Category  string    `json:"Category"`
	CreatedAt time.Time `json:"CreatedAt"`
	UpdatedAt time.Time `json:"UpdatedAt"`
}

// GetID returns ID.
func (dto *PromiseResponse) GetID() string {
	return dto.ID
}

// GetURI returns URI.
func (dto *PromiseResponse) GetURI() string {
	return dto.URI
}

// GetCategory returns Category.
func (dto *PromiseResponse) GetCategory() string {
	return dto.Category
}

// GetCreatedAt returns CreatedAt.
func (dto *PromiseResponse) GetCreatedAt() time.Time {
	return dto.CreatedAt
}

// GetUpdatedAt returns UpdatedAt.
func (dto *PromiseResponse) GetUpdatedAt() time.Time {
	return dto.UpdatedAt
}

// Load data from model.
func (dto *PromiseResponse) Load(m *model.PromiseModel) {
	dto.ID = m.ID
	dto.URI = constValue.CategoryToURI(m.Category, m.ID)
	dto.Category = m.Category
	dto.CreatedAt = m.CreatedAt
	dto.UpdatedAt = m.UpdatedAt
}

File: integration-test/src/test/java/com/promise/integrationtest/dto/PromiseResponse.java
package com.promise.integrationtest.dto;

import com.fasterxml.jackson.annotation.JsonProperty;

public class PromiseResponse
{
    @JsonProperty(value = "ID", required = true)
    private String id;
    @JsonProperty(value = "URI", required = true)
    private String uri;
    @JsonProperty(value = "CreateAt", required = true)
    private String createdAt;
    @JsonProperty(value = "UpdatedAt", required = true)
    private String updatedAt;

    public String getId()
    {
        return id;
    }

    public void setId(String id)
    {
        this.id = id;
    }

    public String getUri()
    {
        return uri;
    }

    public void setUri(String uri)
    {
        this.uri = uri;
    }

    public String getCreatedAt()
    {
        return createdAt;
    }

    public void setCreatedAt(String createdAt)
    {
        this.createdAt = createdAt;
    }

    public String getUpdatedAt()
    {
        return updatedAt;
    }

    public void setUpdatedAt(String updatedAt)
    {
        this.updatedAt = updatedAt;
    }
}
